pkg/registry/pod/strategy.go

```diff
@@ -153,6 +154,23 @@ func (podStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
 	return validation.ValidatePodStatusUpdate(obj.(*api.Pod), old.(*api.Pod))
 }
 
+func CheckPod(r *registry.Store, ctx api.Context, err error, obj runtime.Object) error {
+	if !errors.IsAlreadyExists(err) {
+		return err
+	}
+	pod := obj.(*api.Pod)
+	retObj, getPodErr := r.Get(ctx, pod.Name)
+	if getPodErr != nil {
+		return err
+	}
+	retPod := retObj.(*api.Pod)
+	if api.IsPodTerminated(retPod) {
+		status_err, _ := err.(*errors.StatusError)
+		status_err.ErrStatus.Message += fmt.Sprintf("\n Existant pod is in terminated state! Please check its status and delete it!\n Use `kubectl get pods/%s` \n", pod.Name)
```

Comment:
I don't think that's a sufficient justification for doing it here.  If it's really hard to add this check in a clean way, the check just isn't worth it.